Adding support for references to another peripheral

idubrov opened the pull request:
Handling "[PERIPHERAL].[REGISTER].[FIELD].[ENUMERATEDVALUES]" format
for the "derivedFrom" attribute in <enumeratedValues>.

(used https://www.keil.com/pack/doc/CMSIS/SVD/html/elem_registers.html#elem_enumeratedValues as a reference).

japaric commented on src/generate.rs:
```diff
@@ -767,7 +772,19 @@ pub fn fields(
+                        mod_items.push(
+                            quote! {
+                                #[doc = #description]
+                                pub type #pc_r = super::super::#mod1_::#mod2_::#base_pc_r;
```
nit: I think the super::super is not necessary as all the peripheral modules are in the root of the crate.

japaric commented on src/generate.rs:
```diff
@@ -1058,7 +1076,24 @@ pub fn fields(
+                                    quote! {
+                                        #[doc = #pc_w_doc]
+                                        pub type #pc_w =
+                                            super::super::#mod1_::#mod2_::#base_pc_w;
```
nit: same comment about super::super as above

japaric commented on src/generate.rs:
```diff
@@ -1058,7 +1076,24 @@ pub fn fields(
+                                );
+
+                                quote! {
+                                    super::super::#mod1_::#mod2_::#base_pc_w
```
nit: same comment about super::super as above

japaric commented:
Thanks for the PR @idubrov. This looks good to me. I left a comment about the use of super::super which I think is not necessary.

idubrov commented:
Replaced super::super:: with :: (I'm not sure I had to do it -- my knowledge of Rust is currently at the level of "copy-paste monkey"). Also, renamed mod1/mod2 to _slightly_ more descriptive pmod/rmod.

japaric commented:
Thanks, @idubrov!

@homunkulus r+

P.S. Please next time push a new commit when you do a change instead of rebasing the last one. That makes reviewing easier.

homunkulus commented:
:pushpin: Commit 968fbc5 has been approved by `japaric`

<!-- @homunkulus r=japaric 968fbc5da03974bc9b9956dec4509fc4b03d784a -->

homunkulus commented:
:hourglass: Testing commit 968fbc5 with merge 968fbc5...

homunkulus commented:
:sunny: Test successful - [status-appveyor](https://ci.appveyor.com/project/japaric/svd2rust/build/1.0.75), [status-travis](https://travis-ci.org/japaric/svd2rust/builds/235728776?utm_source=github_status&utm_medium=notification)
Approved by: japaric
Pushing 968fbc5da03974bc9b9956dec4509fc4b03d784a to master...
